ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/StatementExecutorTest.java

```diff
@@ -176,4 +179,97 @@ public void shouldHandlePriorStatement() throws Exception {
     Assert.assertEquals(CommandStatus.Status.ERROR, statusStore.get(ctasCommandId).getStatus());
   }
 
+
+  @Test
+  public void shouldEnforceReferentialIntegrity() throws Exception {
```

Comment:
There is a lot going on in this test. Can we break it up into multiple smaller tests. Also looks like there are at least 3 lines that would break checkstyle

Reply:
I fixed the formatting issues. As for breaking it into multiple tests it is possible but would result in much more code for testing the same cases. Most of the code in this test belongs to creating commands and executing them. It tests both failing when the stream/table is being used and success when stream/table is not being used. I added some comments to clarify it more but if you have any suggestion on how to break it down to multiple tests let me know.

Reply:
I think it would make sense to factor the shared `CTAS` and `CSAS` commands out into a helper method, and then create individual tests for each case, much as you do in the `KsqlEngineTest`.

Reply:
I'f you don't want to break it into multiple tests then i suggest breaking it up into methods. This test method just looks like a wall of text to me.